clients/src/main/java/org/apache/kafka/common/requests/OffsetsForLeaderEpochRequest.java

```diff
@@ -101,23 +107,28 @@ public int replicaId() {
         private final Map<TopicPartition, PartitionData> epochsByPartition;
         private final int replicaId;
 
-        Builder(short version, Map<TopicPartition, PartitionData> epochsByPartition, int replicaId) {
-            super(ApiKeys.OFFSET_FOR_LEADER_EPOCH, version);
+        Builder(short oldestAllowedVersion, short latestAllowedVersion, Map<TopicPartition, PartitionData> epochsByPartition, int replicaId) {
+            super(ApiKeys.OFFSET_FOR_LEADER_EPOCH, oldestAllowedVersion, latestAllowedVersion);
             this.epochsByPartition = epochsByPartition;
             this.replicaId = replicaId;
         }
 
-        public static Builder forConsumer(short version, Map<TopicPartition, PartitionData> epochsByPartition) {
-            return new Builder(version, epochsByPartition, CONSUMER_REPLICA_ID);
+        public static Builder forConsumer(Map<TopicPartition, PartitionData> epochsByPartition) {
+            // Old versions of this API may require CLUSTER authorization which is not typically granted
+            // to clients. We therefore require the latest version in order to be sure that we are not
```

Comment:
"therefore we require the version where this requirement was relaxed" or something? It won't necessarily be the latest version if we add new versions (and this comment won't have to be changed if we rephrase it in this way).